### ISSUE TITLE
Bug: no-implied-eval reports the entire call expression for global references

### ISSUE THREAD

**Issue by TKDev7** [eslint/eslint#19938]
### Environment

Node version:
npm version:
Local ESLint version:
Global ESLint version:
Operating System:

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
{
  "rules": {
    "no-implied-eval": "error"
  }
}
```
</details>

```js
window.setTimeout("count = 5", 10);
```

### What did you expect to happen?

I expected ESLint to report only the method name (e.g., `setTimeout`) in the error, so that only the relevant part of the code is highlighted.

### What actually happened?

ESLint reported the entire call expression, so the whole line was highlighted.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW1wbGllZC1ldmFsOiBcImVycm9yXCIqL1xuLypnbG9iYWwgd2luZG93Ki9cblxud2luZG93LnNldFRpbWVvdXQoXCJjb3VudCA9IDVcIiwgMTApOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

I noticed this while working on #19937. I am not sure whether this is a bug or working as intended.

**Comment by snitin315** [comment]
I feel this is working as expected. It should report the whole call expression.
